scripts/build/dev_generator.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"os"
+)
+
+func init() {
+	buildGenerators["dev"] = devGenerator{}
+}
+
+type devGenerator struct {
+}
+
+// There are a few dev nodes which are required to run an initramfs.
+func (g devGenerator) generate(config Config) ([]file, error) {
```

Comment:
The question is how much to create on boot and how much to embed in the initramfs. Having done this both ways,  I would prefer init to do this on boot, especially for device nodes. One reason is that you're never guaranteed that the initramfs you generate, /dev/ for example,  doesn't get overridden in some way at boot, by future kernel changes, so I'd rather not count on these files being there.  I've been burned every time I've made the decision to pre-create lots of files in the initramfs. Also while we're pretty safe on /dev/null, what if we someday have to look in /sys for other major numbers? It's a bad idea to embed major/minor in something like this. I've gotten hurt by that too.

So I'd really like to make this list much, much smaller. tcz can be removed for sure.